src/transmitter/index.ts

```diff
@@ -265,13 +265,19 @@ export async function retryRequest(
   }
 
   let response: NeedleResponse | undefined;
-  let attempt: number;
 
-  for (attempt = 1; attempt <= retry.attempts; attempt++) {
+  let attempt = 0;
+  while (true) {
+    attempt++;
     const stillHaveRetries = attempt + 1 <= retry.attempts;
     try {
       response = await needle(verb, url, payload, options);
-      if (!(response.statusCode === 502 && stillHaveRetries)) {
+      if ([429, 502, 503, 504].includes(response?.statusCode || 0)) {
+        await sleep(retry.intervalSeconds * 1000);
```

Comment:
if our biggest cluster has 10,000 workloads, will this be enough to get all the requests through at 1,700/sec?

does the queue help with this?

Reply:
Sounds like it would take at least 6 minutes

Btw https://apidocs.snyk.io/?version=2023-01-30%7Ebeta#overview lists an even worse limit of 540rpm, so at least 18 minutes

(that's all ignoring the scan results requests also happening)